### PR TITLE
Changes from $request->get() to correct methods

### PR DESCRIPTION
**Pull request by adudekyeti**
Changes from $request->get() to correct methods

### CODE REVIEW

File: modules/Settings/Roles/actions/Delete.php [hunk 1]
```diff
@@ -13,8 +13,8 @@ class Settings_Roles_Delete_Action extends Settings_Vtiger_Basic_Action
 	public function process(\App\Request $request)
 	{
 		$qualifiedModuleName = $request->getModule(false);
-		$recordId = $request->get('record');
-		$transferRecordId = $request->get('transfer_record');
+		$recordId = $request->getInteger('record');
```
Comment (TomaszQr) [inline modules/Settings/Roles/actions/Delete.php]: not getInteger but  $request->getByType('record', 'Alnum')

File: modules/Settings/Roles/actions/Delete.php [hunk 2]
```diff
@@ -13,8 +13,8 @@ class Settings_Roles_Delete_Action extends Settings_Vtiger_Basic_Action
...
+		$transferRecordId = $request->getInteger('transfer_record');
```
Comment (TomaszQr) [inline modules/Settings/Roles/actions/Delete.php]: not getInteger but  $request->getByType('transfer_record', 'Alnum')

File: modules/Settings/Github/actions/SaveKeysAjax.php [hunk 3]
```diff
@@ -16,7 +16,7 @@ public function process(\App\Request $request)
 	{
 		$clientModel = Settings_Github_Client_Model::getInstance();
 		$clientModel->setToken($request->getByType('token', 'Alnum'));
-		$clientModel->setUsername($request->get('username'));
+		$clientModel->setUsername($request->getByType('username', 'Alnum'));
```
Comment (TomaszQr) [inline modules/Settings/Github/actions/SaveKeysAjax.php]: Text ?

File: modules/Settings/WebserviceApps/views/CreateApp.php [hunk 4]
```diff
@@ -19,7 +19,7 @@ public function process(\App\Request $request)
 		parent::preProcess($request);
 		$moduleName = $request->getModule();
 		$qualifiedModuleName = $request->getModule(false);
-		$recordId = $request->get('record');
+		$recordId = $request->getInteger('record');
 		if (!empty($recordId)) {
```
Comment (mariuszkrzaczkowski) [inline modules/Settings/WebserviceApps/views/CreateApp.php]: has or isEmpty

File: modules/Settings/Workflows/views/Edit.php [hunk 5]
```diff
@@ -25,7 +25,7 @@ public function preProcess(\App\Request $request, $display = true)
 		parent::preProcess($request);
 		$viewer = $this->getViewer($request);
 
-		$recordId = $request->get('record');
+		$recordId = $request->getInteger('record');
 		$viewer->assign('RECORDID', $recordId);
 		if ($recordId) {
```
Comment (mariuszkrzaczkowski) [inline modules/Settings/Workflows/views/Edit.php]: has or isEmpty

File: modules/Settings/Workflows/views/Edit.php [hunk 6]
```diff
@@ -42,7 +42,7 @@ public function step1(\App\Request $request)
 		$qualifiedModuleName = $request->getModule(false);
 		$weekDays = ['Sunday' => 0, 'Monday' => 1, 'Tuesday' => 2, 'Wednesday' => 3, 'Thursday' => 4, 'Friday' => 5, 'Saturday' => 6];
 
-		$recordId = $request->get('record');
+		$recordId = $request->getInteger('record');
 		if ($recordId) {
```
Comment (mariuszkrzaczkowski) [inline modules/Settings/Workflows/views/Edit.php]: ?

File: modules/Settings/Workflows/views/Edit.php [hunk 7]
```diff
@@ -76,7 +76,7 @@ public function step2(\App\Request $request)
 		$moduleName = $request->getModule();
 		$qualifiedModuleName = $request->getModule(false);
 
-		$recordId = $request->get('record');
+		$recordId = $request->getInteger('record');
 
 		if ($recordId) {
```
Comment (mariuszkrzaczkowski) [inline modules/Settings/Workflows/views/Edit.php]: ?

File: modules/Settings/Workflows/views/Edit.php [hunk 8]
```diff
@@ -131,7 +131,7 @@ public function step3(\App\Request $request)
...
```
Comment (mariuszkrzaczkowski) [inline modules/Settings/Workflows/views/Edit.php]: ?

File: modules/Users/views/PreferenceEdit.php [hunk 9]
```diff
@@ -57,7 +57,7 @@ protected function preProcessDisplay(\App\Request $request)
 	public function process(\App\Request $request)
 	{
 		$moduleName = $request->getModule();
-		$recordId = $request->get('record');
+		$recordId = $request->getInteger('record');
 		if (!empty($recordId)) {
```
Comment (mariuszkrzaczkowski) [inline modules/Users/views/PreferenceEdit.php]: ?

File: modules/Vtiger/views/AutoAssignRecord.php [hunk 10]
```diff
@@ -20,7 +20,7 @@ class Vtiger_AutoAssignRecord_View extends Vtiger_BasicModal_View
 	 */
 	public function checkPermission(\App\Request $request)
 	{
-		$recordId = $request->get('record');
+		$recordId = $request->getInteger('record');
 		if (!empty($recordId)) {
```
Comment (mariuszkrzaczkowski) [inline modules/Vtiger/views/AutoAssignRecord.php]: ?

File: modules/Settings/MailSmtp/views/Edit.php [hunk 11]
```diff
@@ -18,8 +18,8 @@ public function process(\App\Request $request)
 	{
 		$moduleName = $request->getModule(false);
 		$viewer = $this->getViewer($request);
-		$record = $request->get('record');
-		if (!empty($record)) {
+		$recordId = !$request->isEmpty('record') ? $request->getInteger('record') : '';
```
Comment (rskrzypczak) [inline modules/Settings/MailSmtp/views/Edit.php]:
```suggestion
		$record = !$request->isEmpty('record') ? $request->getInteger('record') : '';
```

File: modules/Settings/MailSmtp/views/Edit.php [hunk 12]
```diff
@@ -18,8 +18,8 @@ public function process(\App\Request $request)
...
+		if ($recordId) {
```
Comment (rskrzypczak) [inline modules/Settings/MailSmtp/views/Edit.php]:
```suggestion
		if ($record) {
```

File: modules/Settings/CurrencyUpdate/models/Module.php [hunk 13]
```diff
@@ -180,7 +180,7 @@ public function getRatesHistory($bankId, $dateCur, \App\Request $request)
 			->innerJoin('yetiforce_currencyupdate_banks', 'yetiforce_currencyupdate_banks.id = yetiforce_currencyupdate.bank_id')
 			->where(['yetiforce_currencyupdate.bank_id' => $bankId]);
 		// filter by date - if not exists then display this months history
-		if ($request->get('duedate') == '' && $dateCur) {
+		if ($request->getRaw('duedate') === '' && $dateCur) {
```
Comment (rskrzypczak) [inline modules/Settings/CurrencyUpdate/models/Module.php]: isEmpty

**Pull request merged**